plugins/coverimage.koplugin/main.lua

```diff
@@ -0,0 +1,134 @@
+local Device = require("device")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local logger = require("logger")
+local _ = require("gettext")
+local T = require("ffi/util").template
+
+local CoverImage = WidgetContainer:new{
+    name = 'coverimage',
+    is_doc_only = true,
+    settings_id = 0,
+}
+
+function CoverImage:_enabled()
+    return self.enabled
+end
+
+function CoverImage:_restore()
+    return self.enabled and self.restore
+end
+
+function CoverImage:init()
+    self.cover_image_path = G_reader_settings:readSetting("cover_image_path") or "Cover.jpg"
+    self.enabled = G_reader_settings:isTrue("cover_image_enabled")
+    self.restore = G_reader_settings:isTrue("cover_image_restore") and self.enabled
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function CoverImage:onCloseDocument()
+    logger.dbg("CoverImage: onCloseDocument")
+    local lfs = require("libs/libkoreader-lfs")
+    -- Try to restore the state before start of KOReader.
+    -- If KOReader gets updated during operation (at least on Android) the backup survives.
+    -- Under normal conditions this should not happen.
+    if self.restore then
+        -- Delete image unconditionally, so if no backup exists, there will be no cover file.
+        -- This is useful on Tolino to use the system sleep image after KOReader exit.
+        if lfs.attributes(self.cover_image_path) then
+            os.execute("rm " .. self.cover_image_path)
+        end
+        -- Restore backup if it exists, so we can use the last image.
+        -- On Tolino a user defined /sdcard/tolino_others.jgp can be used as screensaver on system sleep.
+        if lfs.attributes(self.cover_image_path ..".bak") then
+            os.execute("mv " .. self.cover_image_path .. ".bak " .. self.cover_image_path)
+        end
+    end
+end
+
+function CoverImage:onDocSettingsLoad(doc_settings, document)
```

Comment:
In `onReaderReady` I have added:
```
function CoverImage:onReaderReady(doc_settings)
   logger.dbg("CoverImage: onDocSettingsLoad")
   if self.enabled and self.ui and self.ui.document then 
        local lfs = require("libs/libkoreader-lfs")
```

The plugin is not shown in FileManager, because I have set `    is_doc_only = true,`